tests/test_runner.py

```diff
@@ -1,18 +1,29 @@
 import os
 
+import pytest
+
+from nnbench.context import cpuarch, python_version, system
 from nnbench.runner import BenchmarkRunner
 
 
-def test_runner_discovery(testfolder: str) -> None:
+def test_runner_discovery(testfolder: str, another_testfolder: str) -> None:
     r = BenchmarkRunner()
 
     r.collect(os.path.join(testfolder, "standard_benchmarks.py"), tags=("runner-collect",))
     assert len(r.benchmarks) == 1
-
     r.clear()
 
     r.collect(testfolder, tags=("runner-collect",))
     assert len(r.benchmarks) == 1
+    r.clear()
+
+    r.collect(testfolder, tags=("non-existing-tag",))
+    assert len(r.benchmarks) == 0
+    r.clear()
+
+    r.collect(os.path.join(testfolder), tags=("runner-collect",))
+    r.collect(os.path.join(another_testfolder), tags=("runner-collect",))
+    assert len(r.benchmarks) == 2
```

Comment:
I think you can `assert len(r.benchmarks) == 1` after the first collect and inline L16-L18 here.